crates/services/sync/src/import.rs

```diff
@@ -300,68 +361,76 @@ fn get_block_stream<
+                    let checked_headers = results
+                        .into_iter()
+                        .take_while(|header| {
+                            check_sealed_header(header, peer.clone(), &p2p, &consensus)
+                        })
+                        .collect::<Vec<_>>();
+                    Batch::new(peer, range, checked_headers)
+                }
             }
         })
-        .map(move |headers| {
-            let consensus = consensus.clone();
+        .map({
             let p2p = p2p.clone();
-            async move {
-                let Batch {
-                    peer,
-                    range,
-                    results,
-                } = headers;
-                if results.is_empty() {
-                    SealedBlockBatch::new(peer, range, vec![])
-                } else {
-                    await_da_height(
-                        results
-                            .last()
-                            .expect("We checked headers are not empty above"),
-                        &consensus,
-                    )
-                    .await;
-                    let headers = SealedHeaderBatch::new(peer, range, results);
-                    get_blocks(&p2p, headers).await
+            let consensus = consensus.clone();
+            move |headers| {
+                let p2p = p2p.clone();
+                let consensus = consensus.clone();
+                async move {
+                    let Batch {
+                        peer,
+                        range,
+                        results,
+                    } = headers.await;
+                    if results.is_empty() {
+                        SealedBlockBatch::new(peer, range, vec![])
+                    } else {
+                        await_da_height(
+                            results
+                                .last()
+                                .expect("We checked headers are not empty above"),
+                            &consensus,
+                        )
+                        .await;
+                        let headers = SealedHeaderBatch::new(peer, range, results);
+                        get_blocks(&p2p, headers).await
+                    }
                 }
+                .instrument(tracing::debug_span!("consensus_and_transactions"))
+                .in_current_span()
             }
-            .instrument(tracing::debug_span!("consensus_and_transactions"))
-            .in_current_span()
         })
 }
 
 fn get_header_batch_stream<P: PeerToPeerPort + Send + Sync + 'static>(
     range: RangeInclusive<u32>,
-    params: &Config,
+    params: Config,
     p2p: Arc<P>,
-) -> impl Stream<Item = SealedHeaderBatch> {
+) -> impl Stream<Item = impl Future<Output = SealedHeaderBatch>> {
     let Config {
         header_batch_size, ..
     } = params;
-    let ranges = range_chunks(range, *header_batch_size);
-    futures::stream::iter(ranges).then(move |range| {
```

Comment:
So was `then` causing the inner inner future to get awaited? Was this blocking subsequent items in the stream?